src/test/java/uk/gov/companieshouse/model/validation/CompanyNumberValidatorTest.java

```diff
@@ -0,0 +1,74 @@
+package uk.gov.companieshouse.model.validation;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import uk.gov.companieshouse.model.validator.CompanyNumberValidator;
+
+@ExtendWith(SpringExtension.class)
+public class CompanyNumberValidatorTest {
+
+    @Test
+    void shouldThrowAnExceptionWhenInputListIsInWrongFormat() {
+
+        List<String> invalidLists = List.of(",", "A,[.", "A.BC.D");
+
+        invalidLists.forEach(prefixList -> {
```

Comment:
this can be simplified: 
`invalidLists.forEach(prefixList -> assertThrows(IllegalArgumentException.class, () -> new CompanyNumberValidator(prefixList), prefixList));`